File: frontend/src/types/halfway.ts
import type { DashboardActivityItem } from "./admin-dashboard";
import type { OrderSummary } from "./orders";

export interface HalfwaySummary {
  orders_today_count: number;
  active_queue_count: number;
  queue_orders: OrderSummary[];
  recent_transactions: DashboardActivityItem[];
}

File: backend/app/schemas/halfway.py
from __future__ import annotations

from pydantic import BaseModel

from app.schemas.admin_dashboard import DashboardActivityItem
from app.schemas.orders import OrderSummaryResponse


class HalfwaySummaryResponse(BaseModel):
    orders_today_count: int
    active_queue_count: int
    queue_orders: list[OrderSummaryResponse]
    recent_transactions: list[DashboardActivityItem]
